dev-tools/omdb/src/bin/omdb/db.rs

```diff
@@ -3212,6 +3225,10 @@ async fn cmd_db_instances(
         query = query.filter(dsl::time_deleted.is_null());
     }
 
+    if !states.is_empty() {
+        query = query.filter(dsl::state.eq_any(states.clone()));
```

Comment:
`--running` checks for instances in the `Running` effective state, but `--states` checks only the instance's state (and only accepts the internal state variants as parameter values). Should `--states` check effective states instead? My WAG is that we'd be more likely to find this useful (e.g. this way we can search for instances that appear to be stuck in `Starting` or `Stopping` instead of just having to search for those in `Vmm`). Maybe that would be better served by adding `omdb db vmms`, though. WDYT?

Reply:
Hmm. I considered making this filter on effective states instead, but it felt like there was a bit of an unfortunate choice that had to be made in order to do so. 

We could try to push the filtering into the DB, requiring us to express the combination(s) of DB instance and VMM states that represent an effective state in OMDB, which I didn't love, since it felt like we could easily allow OMDB to get out of sync with the actual Nexus behavior, effectively lying to the user. Alternatively, we could query the DB for all instances, and then filter them in memory by checking the effective state, the way that `--running` does. But, I didn't love that, because the `--fetch-limit` flag is implemented by adding a limit to the query, which means that if you request a particular state with `--fetch-limit 100`, and there are only 3 instances in that state in the first 100 instances returned, the command only gives you 3 instances back. Which I don't really love. 

I'm happy to reevaluate these tradeoffs, but I felt personally okay with querying instances by their Nexus instance states...I suppose we could rename this flag to `--nexus-state` or something, and then go back and add a `--effective-state` or something? I dunno

Reply:
Fair point re the application of the `--fetch-limit` flag. I would be on board with saying that `--states` is meant primarily to check for `Creating` and `Failed` instances, and if users want more detail about instances that are in some other effective state, they can/should be looking at VMMs instead (I see we now have #6928 for this--thanks!).

Reply:
I'm happy to rename it to `--nexus-states` if you think that's worth it?

Reply:
Nah, I think `--states` is actually better, because it extends more naturally to other objects (VMMs and disks come immediately to mind) that don't distinguish between "Nexus state" and "effective external state."

(Thinking about it more has also soured me on the idea of making this flag operate on effective states--ISTM the point of a lot of `omdb db` commands is to provide a nicer interface for inspecting database state that you'd otherwise have to get from a CRDB shell; our instance DB queries don't deal with effective external states, so it doesn't make sense to try to work with those here. If anything I'm probably now closer to saying that we should do something else with `--running`, like translating it to `--states vmm`, but I'm totally OK with leaving it alone in this PR.)

Reply:
@gjcolombo re:
> Thinking about it more has also soured me on the idea of making this flag operate on effective states--ISTM the point of a lot of omdb db commands is to provide a nicer interface for inspecting database state that you'd otherwise have to get from a CRDB shell; our instance DB queries don't deal with effective external states, so it doesn't make sense to try to work with those here. 

Yeah, I broadly agree with this. My preference is for the `omdb db` commands to be as "faithful" to the DB models as possible and not hide stuff in the DB from the user. I think including the effective state in `db instance show` is good, for example, but we need to also expose the "real" underlying db representation. 

Incidentally, I'd kinda like to change the `db instances`/`db instance list` command to show columns for the instance state and VMM state, instead of or in addition to the effective state. The only reason I've not done that yet is because the lines that command outputs are already way too long and wrap in almost every terminal.

I'm considering adding a mode to the instance list commands that abandons the single line tabular output and instead shows the instance on one line and the associated VMM on the next line, with indentation or something to show that they're connected. Then we could add way more state information to the output.